openfisca_core/parameters.py

```diff
@@ -117,13 +117,20 @@ def __init__(self, name, data, file_path = None):
         self.name = name
         self.file_path = file_path
         _validate_parameter(self, data, data_type = dict)
-        self.values_history = self  # Only for retro-compatibility
+        self.description = None
+        self.metadata = {}
+        self.values_history = self  # Only for backward compatibility
 
+        # If metadata have been provided
         if data.get('values'):
-            _validate_parameter(self, data, allowed_keys = set(['values', 'description', 'unit', 'reference']))
+            # 'unit' and 'reference' are only listed here for backward compatibility
+            _validate_parameter(self, data, allowed_keys = set(['values', 'description', 'metadata', 'unit', 'reference']))
```

Comment:
Copied from above

The idea is to do a minimal change to the actual situation.
And description can be thought as standard/generic and informative enough
to be placed outside metadata which is more for custom/non-vital metadata.
But @fpagnoux may have more details

Reply:
I actually hesitated while implementing.

The main argument that push to let `description` at the top level is that in [Matti's comment](https://github.com/openfisca/openfisca-core/pull/681#issuecomment-404348529), `description` seem to perfectly fit the definition of a "fully-specified and normalised metadata":

- We have no interrogation about it, and we use it in many places, including on our [`parameters/`](https://fr.openfisca.org/api/v21/parameters) API route.
- `parameter.description` has always been exposed

Now, if `description` is at top level, why not `reference`?

It's arguable, but if everyone is convinced that references are needed, it seems like we are not fully sure about what a reference should _be_ exactly.  And whenever we get a better idea about that, moving `reference` from `metadata` to top level is not even a breaking change for Core.

Reply:
My main issue is constistency between where the description is and where the references are.
Would it be a big issue to break `core` and change `parameter.descritption` to  `parameter.metadata.descritption` ?

Reply:
I understand the argument, but if we want to have both "normalized and standard" and "free custom" metadata, it's hard to have both of them in the same place:
`parameter.metadata['description']` would not  look very normalized 

Do you disagree with the general dichotomy of normalized vs custom?

Reply:
I don't disagree with the paradigm, I think it is interesting to leave room for growth.
However, I think we need to create some kind of guidelines in the documentation for new comers and a timeline on how and when things should be normalized so that we can count on some formating standard to build cool tools that every country can use.

Reply:
I'd rather have normalisation usage-based over time-based 🙂